### PR TITLE
fix(hook): fix branch and hash on pullrequest event + add docs

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: docs/content/docs/concepts/variables.md [hunk 1]
```diff
@@ -97,15 +97,46 @@ echo $CDS_PARENT_APPLICATION
...
+- `{{.git.branch}}`: 
+  - Push event: Name of the branch where the push occured
+  - PullRequest event: Name of the source branch
+- `{{.git.tag}}`: Name of the tag that triggered the run
+- `{{.git.author}}`: Name of the most recent commit author
+- `{{.git.author.email}}`: Email of the most recent commit author
+- `{{.git.message}}`: Git message of the most recent commit
+- `{{.git.server}}`: Name of the repository manager
+- `{{.git.repository}}`: 
+  - Push event:  Name of the repository
```
Comment (yesnault) [inline docs/content/docs/concepts/variables.md:115]: not a variable?

File: engine/hooks/bitbucket_server.go [hunk 2]
```diff
@@ -119,11 +119,11 @@ func getVariableFromBitbucketServerPullRequest(payload map[string]interface{}, p
...
-	payload[GIT_BRANCH_BEFORE] = pr.FromRef.DisplayID
-	payload[GIT_HASH_BEFORE] = pr.FromRef.LatestCommit
-	payload[GIT_BRANCH] = pr.ToRef.DisplayID
-	payload[GIT_HASH] = pr.ToRef.LatestCommit
-	hashShort := pr.ToRef.LatestCommit
+	payload[GIT_BRANCH] = pr.FromRef.DisplayID
+	payload[GIT_HASH] = pr.FromRef.LatestCommit
+	payload[GIT_BRANCH_DEST] = pr.ToRef.DisplayID
+	payload[GIT_HASH_DEST] = pr.ToRef.LatestCommit
+	hashShort := pr.FromRef.LatestCommit
```
Comment (yesnault) [inline engine/hooks/bitbucket_server.go:126]: why FromRef and not ToRet?

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12912.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: docs/content/docs/concepts/variables.md (hunk 1 above)
Comment (sguiheux) [reply docs/content/docs/concepts/variables.md:115]: it's a list :)

**Pull request merged**